Unit test of amplitude estimation fails

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues -->

### Information

- **Qiskit Aqua version**: master (189536b)
- **Qiskit Terra version**: master (https://github.com/Qiskit/qiskit-terra/commit/21f78dc5ee236d51697e65db916ccd04f4dfc3bc)
- **Python version**: 3.8.7
- **Operating system**: macOS 10.15.7

### What is the current behavior?

The unit test of amplitude estimation in Aqua fails.

### Steps to reproduce the problem

```bash
$ python -m unittest test/aqua/test_amplitude_estimation.py
...................F.....F..
======================================================================
FAIL: test_confidence_intervals_1 (test.aqua.test_amplitude_estimation.TestSineIntegral)
End-to-end test for all confidence intervals.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/ima/envs/dev38/lib/python3.8/site-packages/ddt.py", line 182, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/ima/tasks/1_2021/qiskit/aqua/test/aqua/test_amplitude_estimation.py", line 422, in test_confidence_intervals
    self.assertEqual(confint, expected_confint)
AssertionError: Lists differ: [0.7006496745630342, 0.7516435372256463] != [0.24947346406470136, 0.3003771197734433]

First differing element 0:
0.7006496745630342
0.24947346406470136

- [0.7006496745630342, 0.7516435372256463]
+ [0.24947346406470136, 0.3003771197734433]

======================================================================
FAIL: test_statevector_1 (test.aqua.test_amplitude_estimation.TestSineIntegral)
Statevector end-to-end test
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/ima/envs/dev38/lib/python3.8/site-packages/ddt.py", line 182, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/ima/tasks/1_2021/qiskit/aqua/test/aqua/test_amplitude_estimation.py", line 368, in test_statevector
    self.assertAlmostEqual(value, getattr(result, key), places=3,
AssertionError: 0.27029 != 0.7297097034752369 within 3 places (0.4594197034752369 difference) : estimate `mle` failed

----------------------------------------------------------------------
Ran 28 tests in 13.609s

FAILED (failures=2)
```

### What is the expected behavior?

The unit test passes

### Suggested solutions

--- a/qiskit/aqua/algorithms/amplitude_estimators/ae.py
+++ b/qiskit/aqua/algorithms/amplitude_estimators/ae.py
@@ -1,6 +1,6 @@
 # This code is part of Qiskit.
 #
-# (C) Copyright IBM 2018, 2020.
+# (C) Copyright IBM 2018, 2021.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory

--- a/qiskit/aqua/algorithms/amplitude_estimators/ae_algorithm.py
+++ b/qiskit/aqua/algorithms/amplitude_estimators/ae_algorithm.py
@@ -1,6 +1,6 @@
 # This code is part of Qiskit.
 #
-# (C) Copyright IBM 2018, 2020.
+# (C) Copyright IBM 2018, 2021.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory

--- a/qiskit/aqua/algorithms/amplitude_estimators/ae_algorithm.py
+++ b/qiskit/aqua/algorithms/amplitude_estimators/ae_algorithm.py
@@ -158,14 +158,12 @@ class AmplitudeEstimationAlgorithm(QuantumAlgorithm):
                 - self.state_preparation.num_ancillas
 
             oracle = QuantumCircuit(num_state_qubits)
-            oracle.x(self.objective_qubits)
             oracle.h(self.objective_qubits[-1])
             if len(self.objective_qubits) == 1:
                 oracle.x(self.objective_qubits[0])
             else:
                 oracle.mcx(self.objective_qubits[:-1], self.objective_qubits[-1])
             oracle.h(self.objective_qubits[-1])
-            oracle.x(self.objective_qubits)
 
             # construct the grover operator
             return GroverOperator(oracle, self.state_preparation)

--- a/test/aqua/test_amplitude_estimation.py
+++ b/test/aqua/test_amplitude_estimation.py
@@ -1,6 +1,6 @@
 # This code is part of Qiskit.
 #
-# (C) Copyright IBM 2018, 2020.
+# (C) Copyright IBM 2018, 2021.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory

--- a/test/aqua/test_amplitude_estimation.py
+++ b/test/aqua/test_amplitude_estimation.py
@@ -165,9 +165,7 @@ class TestBernoulli(QiskitAquaTestCase):
                     circuit.cry(2 * 2 ** power * angle, qr_eval[power], qr_objective[0])
             else:
                 oracle = QuantumCircuit(1)
-                oracle.x(0)
                 oracle.z(0)
-                oracle.x(0)
 
                 state_preparation = QuantumCircuit(1)
                 state_preparation.ry(angle, 0)
@@ -210,9 +208,7 @@ class TestBernoulli(QiskitAquaTestCase):
 
             else:
                 oracle = QuantumCircuit(1)
-                oracle.x(0)
                 oracle.z(0)
-                oracle.x(0)
                 state_preparation = QuantumCircuit(1)
                 state_preparation.ry(angle, 0)
                 grover_op = GroverOperator(oracle, state_preparation)

